packages/labs/ssr-react/src/lib/node/wrap-create-element.ts

```diff
@@ -19,49 +19,50 @@ import type {
  * Wraps the provided `createElement` function to also server render Lit
  * component's shadow DOM.
  */
-export function wrapCreateElement(
+export const wrapCreateElement = (
   // TODO(augustjk) Should the type for this param be more generic to allow
   // non-React alternatives like preact?
   originalCreateElement: typeof ReactCreateElement
-) {
-  return function litPatchedCreateElement<P>(
-    type: ElementType<P>,
-    props: PropsWithChildren<P> | null,
-    ...children: ReactNode[]
-  ): ReactElement {
-    if (isCustomElement(type)) {
-      const {shadowContents, elementAttributes, templateAttributes} =
-        renderCustomElement(type, props);
+) =>
+  originalCreateElement.name !== 'litPatchedCreateElement'
+    ? function litPatchedCreateElement<P>(
```

Comment:
That is a possibility. The name check here is unfortunately not robust, which is why I added a global flag in `@lit-labs/ssr-react/enable-lit-ssr.js`.

I'm not sure how robust this really needs to be. I added it primarily because our integration test runs the monkey-patching and the jsxImportSource modification in the same context, which would be discouraged in practice.

Actually, now that I think about it, it might be better these checks just need to happen at the site of the monkey patching or replacement. Not in the wrapping function itself. I'm going to try that out!

Reply:
I removed the name check all-together. It doesn't actually seem to be meaningful.